r3bbase/R3BFileSource.cxx

```diff
@@ -370,38 +427,49 @@ void R3BFileSource::SetInTree(TTree* tempTree)
     Init();
 }
 
+// ----  Method getrunid -----------------------------------------------------
+Int_t R3BFileSource::getrunid(uint64_t st)
+{
+    for (Int_t i = 0; i < array_size - 1; i++)
```

Comment:
When using `std::vector`, this loop can be replaced with a call to the `find_if` algorithm.

Also, one would assume that the timestamps in the file are sorted from smaller to larger values, correct? Also, the events are processed in increasing order, correct?
If that is true, then one can store the index into the array/vector (called `i` here) that points to the current timestamp. And then in the next call to getrunid start start from the index of the last timestamp without having to go through all previous ones.